src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -270,6 +270,14 @@ public void StartConnectAsync()
                     peer?.Disconnect("Connection timeout");
                 }
             }
+
+            catch (NBitcoin.Protocol.ProtocolException)
+            {
+                this.logger.LogDebug("Peer {0} handshake failed.", peerAddress.Endpoint);
+                peerAddress.SetHandshakeAttempted(this.dateTimeProvider.GetUtcNow(), handshakeSuccessful: false);
+                peer?.Disconnect("Error while handshaking");
+            }
+
             catch (Exception exception)
             {
                 this.logger.LogTrace("Exception occurred while connecting: {0}", exception.ToString());
```

Comment:
And also this should count as failed handshake

Reply:
Updated.